src/mame/vtech/vsmile.cpp

```diff
@@ -217,18 +231,41 @@ void vsmile_state::banked_map(address_map &map)
 
 static INPUT_PORTS_START( vsmile )
 	PORT_START("REGION")
-	PORT_DIPNAME( 0x0f, 0x04, "sysrom Region" )
-	PORT_DIPSETTING(    0x04, "UK/US" )
-	PORT_DIPSETTING(    0x07, "China" )
-	PORT_DIPSETTING(    0x08, "Mexico" )
-	PORT_DIPSETTING(    0x0a, "Italy" ) // not valid on V.Smile Motion?
-	PORT_DIPSETTING(    0x0b, "Germany" )
-	PORT_DIPSETTING(    0x0c, "Spain" )
-	PORT_DIPSETTING(    0x0d, "France" )
+	//based on schematics and BIOS test screen
+	PORT_DIPNAME( 0x0f, 0x04, DEF_STR(Language) )
+	//PORT_DIPSETTING(    0x00, "0" ) //invalid UK
+	//PORT_DIPSETTING(    0x01, "1" ) //invalid US
+	PORT_DIPSETTING(    0x02, DEF_STR(Italian) ) //IT
+	//PORT_DIPSETTING(    0x03, "3" ) //invalid US
+	//PORT_DIPSETTING(    0x04, "4" ) //invalid US
+	//PORT_DIPSETTING(    0x05, "5" ) //invalid US
+	//PORT_DIPSETTING(    0x06, "6" ) //invalid UK
+	PORT_DIPSETTING(    0x07, DEF_STR(Chinese) ) //Chinese
+	PORT_DIPSETTING(    0x08, "Portuguese" ) //PO
+	PORT_DIPSETTING(    0x09, "Dutch" ) //DU
+	//PORT_DIPSETTING(    0x0a, "a" ) //invalid UK
+	PORT_DIPSETTING(    0x0b, DEF_STR(German) ) //GE
+	PORT_DIPSETTING(    0x0c, DEF_STR(Spanish) ) //SP
+	PORT_DIPSETTING(    0x0d, DEF_STR(French) ) //FR
+	PORT_DIPSETTING(    0x0e, "UK" ) //UK
+	PORT_DIPSETTING(    0x0f, "US" ) //US
 	PORT_DIPNAME( 0x10, 0x10, "VTech Intro" )
-	PORT_DIPSETTING(    0x00, "Off" )
-	PORT_DIPSETTING(    0x10, "On" )
+	PORT_DIPSETTING(    0x00, DEF_STR(Off) )
+	PORT_DIPSETTING(    0x10, DEF_STR(On) )
 	PORT_BIT( 0xe0, 0x00, IPT_UNUSED )
+
+	//using dipswitches for OFF and ON buttons to be able to press them at boot
+	PORT_START("SYSTEM")
+	PORT_DIPNAME( 0x40, 0x40, "OFF switch" 	/*VSMILE_PORTB_OFF_SW*/)
+	PORT_DIPSETTING(    0x00, DEF_STR(On) )
+	PORT_DIPSETTING(    0x40, DEF_STR(Off) )
+	PORT_DIPNAME( 0x80, 0x80, "ON switch" 	/*VSMILE_PORTB_ON_SW*/)
+	PORT_DIPSETTING(    0x00, DEF_STR(On) )
+	PORT_DIPSETTING(    0x80, DEF_STR(Off) )
+	PORT_DIPNAME( 0x08, 0x08, "Restart" 	/*VSMILE_PORTB_RESET*/)
+	PORT_DIPSETTING(    0x00, DEF_STR(On) )
+	PORT_DIPSETTING(    0x08, DEF_STR(Off) )
```

Comment:
Restart isn't a button but a jumper on board
I used powerOn PowerOff first, but you have to get them pressed AT START, not when already launched so it's impossible to get them pressed using classic inputs.
If you could point me on another way to do it, I'll be more than happy to fix it and use buttons stuff.
This is what I did first :
```
        PORT_START("SYSTEM")
	PORT_BIT( 0x40 /*VSMILE_PORTB_OFF_SW*/, IP_ACTIVE_LOW, IPT_POWER_OFF )
	PORT_BIT( 0x80 /*VSMILE_PORTB_ON_SW*/, IP_ACTIVE_LOW, IPT_POWER_ON )
	PORT_BIT( 0x08 /*VSMILE_PORTB_RESET*/, IP_ACTIVE_LOW, IPT_SERVICE )
	PORT_BIT( 0x37, 0x00, IPT_UNUSED )
```

Reply:
Board jumpers should be implemented as machine configuration switches, not DIP switches, in any case.

Reply:
Also, it's perfectly possible to have them held at start-up, that's what the F3 key is for. Just implement the jumpers as machine configuration switches rather than DIPs, make the buttons actual buttons, and don't mangle the `PORT_BIT` definitions with inline comments like that.

The fact that the port is wired up to port B on the SoC can be inferred from looking at the code that reads the port, the fact that they're on/off buttons is implied by the use of `IPT_POWER_OFF` and `IPT_POWER_ON`, and the fact that it's a V.Smile is clear from it being in vsmile.cpp, so those three inline comments are unnecessary.

Reply:
I presume that the problem here is that the machine expects the power-on button to be pressed while it's powering on. Hence the power-on 'dip switch' is default to on.
Then, if it was an emulated button, a user wouldnt be able to start emulation unless by awkwardly holding F1 and then pressing F3.